lib/galaxy/webapps/galaxy/services/history_contents.py

```diff
@@ -1600,9 +1609,14 @@ def _purge(self, item: HistoryItemModel, trans: ProvidesHistoryContext):
             return self.dataset_collection_manager.delete(trans, "history", item.id, recursive=True, purge=True)
         self.hda_manager.purge(item, flush=self.flush)
 
-    def _change_datatype(self, item: HistoryItemModel, params: ChangeDatatypeOperationParams):
+    def _change_datatype(
+        self, item: HistoryItemModel, params: ChangeDatatypeOperationParams, trans: ProvidesHistoryContext
+    ):
         if isinstance(item, HistoryDatasetAssociation):
-            item.change_datatype(params.datatype)
+            if params.datatype == "auto":  # Trigger job to auto-detect datatype
+                self.hda_manager.set_metadata(trans, item, overwrite=True)
+            else:
+                set_metadata.delay(self.hda_manager, None, dataset_id=item.id)
```

Comment:
Is this the correct way of regenerating the metadata? It seems I'm not doing it right... cause I get `Object of type HDAManager is not JSON serializable` :thinking: